ui/admin/app/templates/scopes/scope/host-catalogs/host-catalog/host-sets/host-set/hosts/index.hbs

```diff
@@ -28,82 +28,80 @@
 
   <page.body>
     {{#if @model.hosts}}
-      <Rose::Table as |table|>
-        <table.header as |header|>
-          <header.row as |row|>
-            {{#if @model.hostSet.isStatic}}
-              <row.headerCell>{{t 'form.name.label'}}</row.headerCell>
-            {{else}}
-              <row.headerCell>{{t 'form.external_name.label'}}</row.headerCell>
-            {{/if}}
-            <row.headerCell>{{t 'form.type.label'}}</row.headerCell>
-            <row.headerCell>{{t 'form.id.label'}}</row.headerCell>
-            {{#if @model.hostSet.isStatic}}
-              <row.headerCell>{{t 'titles.actions'}}</row.headerCell>
-            {{/if}}
-          </header.row>
-        </table.header>
+      <Hds::Table
+        @valign='middle'
+        @model={{@model.hosts}}
+        @columns={{(if
+          @model.hostSet.isStatic
+          (array
+            (hash label=(t 'form.name.label'))
+            (hash label=(t 'form.type.label'))
+            (hash label=(t 'form.id.label'))
+            (hash label=(t 'titles.actions'))
+          )
+          (array
+            (hash label=(t 'form.external_name.label'))
+            (hash label=(t 'form.type.label'))
+            (hash label=(t 'form.id.label'))
+          )
+        )}}
+      >
 
-        <table.body as |body|>
-          {{#each @model.hosts as |host|}}
-            <body.row as |row|>
-              <row.headerCell>
-                {{#if host.isStatic}}
-                  <LinkTo
-                    @route='scopes.scope.host-catalogs.host-catalog.hosts.host'
-                    @model={{host.id}}
-                  >
-                    {{host.displayName}}
-                  </LinkTo>
-                {{else}}
-                  {{! Link to a different subroute for dynamic/plugin hosts }}
-                  <LinkTo
-                    @route='scopes.scope.host-catalogs.host-catalog.host-sets.host-set.hosts.host'
-                    @model={{host.id}}
-                  >
-                    {{host.displayName}}
-                  </LinkTo>
-                {{/if}}
-                <p>{{host.description}}</p>
-              </row.headerCell>
-              <row.cell>
-                <HostCatalogTypeBadge @model={{host}} />
-              </row.cell>
-              <row.cell>
-                <Copyable
-                  @text={{host.id}}
-                  @buttonText={{t 'actions.copy-to-clipboard'}}
-                  @acknowledgeText={{t 'states.copied'}}
+        <:body as |B|>
+          <B.Tr>
+            <B.Td>
+              {{#if B.data.isStatic}}
```

Comment:
@dvonanderson 
Wondering if this conditional is actually needed. If a host set is dynamic (plugin, azure or aws), can the host set have hosts that are static? and vice versa?

Reply:
@calcaide a question for me is why are we providing a sub-route for dynamic/plugin host?  I removed the conditional and worked fine for the  ` @route='scopes.scope.host-catalogs.host-catalog.host-sets.host-set.hosts.host'`

Reply:
@dvonanderson 
Yep, that's why we need to first validate the question of: Can a host-set with specific type (static, plugin, aws, azure) contain hosts that have different types that the host-set?

This questions is something we need to figure out first, once we know this, will be easier to reason about this case.

Good call on the sub-route, perhaps you can use git history and see when we add this change, usually that brings context. Need to figure out these two things before.

Reply:
@calcaide resolving this conversation

Reply:
@dvonanderson Resolving this conversation which what outcome? Can you provide an answer to the question stated above?

Reply:
@calcaide yep sorry thought that was more of a search and discovery point, but this change was part of template refactor to deprecate `renderTemplate`

Reply:
@dvonanderson 

I am not asking about when we added this, I can check git history, and yeah, this was part of refactor `renderTemplate` 15 months ago.

What I am still looking for is an answer to the question which I started this conversation, [here the link](https://github.com/hashicorp/boundary-ui/pull/2049#discussion_r1435239638), here the question: `If a host set is dynamic (plugin, azure or aws), can the host set have hosts that are static? and vice versa?`


In my [message here](https://github.com/hashicorp/boundary-ui/pull/2049#discussion_r1435304081) it was saying that first, we need to answer this question. I don't think is a search & discovery point, we are refactoring/migrating tables, this conditional is part of the table, I am not sure is need, in case is not, we are computing a conditional several times for no reason. Since I am not sure (from the domain PoV) this conditional is need, I would like to know that first.

I will appreciate an answer to such question

Reply:
@calcaide reached out to the BE team, but Dharini was there to answer the question. Host sets have to match the type of host set.   https://hashicorp.slack.com/archives/C014AHXUL3G/p1704482262367159

Reply:
@dvonanderson 
Ok. So what we found out is that we have 2 types and are exclusive. So or is type A or is type B.

For that reason, we have 2 table types. I do suggest to:
1. Find in our codebase we have a pattern for templates that have exclusive table types. Like [this one](https://github.com/hashicorp/boundary-ui/blob/main/ui/admin/app/templates/scopes/scope/auth-methods/auth-method/accounts/index.hbs#L25-L168) or [this one](https://github.com/hashicorp/boundary-ui/blob/main/ui/admin/app/templates/scopes/scope/auth-methods/auth-method/managed-groups/index.hbs#L26-L109).
2. Follow the current pattern for this case. If you find the current pattern can be improved, has flaws or can't fit your specific case let's discuss. I strongly recommend leverage tech time so we get team consensus. But feel free to ping me if you want to discuss further.

Why I suggest point 1? If the type is exclusive, we should check the condition just once. If you are type A, render table for type A. If you are type B, render table for type B. Then, the only conditionals left within the table (if any) should be related to permissions.

Thoughts?

Reply:
@calcaide we discussed in last tech time.  I will ping you tomorrow

Reply:
@calcaide I just pushed a commit that adheres to our current pattern take a look and give me you thoughts?